src/main/java/BallVision.java

```diff
@@ -10,107 +11,234 @@ Basic vision algorithm for detecting balls (powercells)
+                new Size(matContourPair.mat.width(), matContourPair.mat.height()),
+                Math.toRadians(68.5),
+                16.0,
+                9.0
+        );
+
+        return new TargetLocation(
+                best.x,
+                best.y,
+                best.width,
+                best.height,
+                yaw,
+                0 // pitch is irrelevant so just pass 0
+        );
     }
 
     /* calculate yaw or pitch (in radians) */
     static double calcAngle(long val, long centerVal, double focalLen) {
-        return Math.atan(((double)(val - centerVal)) / focalLen);
+        return Math.atan(((double) (val - centerVal)) / focalLen);
     }
 
-    public static Mat algorithm(Mat input) {
+    public static MatContourPair ballDetectPipeline(Mat input) {
         // Resize
         Mat resizeInput = input;
         double resizeWidth = 320.0;
-        double resizeHeight = 240.0;
+        double resizeHeight = 181.3;
         Size toScale = new Size(resizeWidth, resizeHeight);
         Mat resizedOutput = new Mat();
         Imgproc.resize(resizeInput, resizedOutput, toScale);
 
-        /* calculate camera information
-        double aspectDiag = Math.hypot(aspectH, aspectV);
-
-        double fieldViewH = Math.atan(Math.tan(diagFieldView / 2.0) * (aspectH / aspectDiag)) * 2.0;
-        double fieldViewV = Math.atan(Math.tan(diagFieldView / 2.0) * (aspectV / aspectDiag)) * 2.0;
-
-        double hFocalLen = resizedOutput.width / (2.0 * Math.tan((fieldViewH / 2.0)));
-        double vFocalLen = resizedOutput.height / (2.0 * Math.tan((fieldViewV / 2.0)));
-
-         */
         // Blur
         Mat blurInput = resizedOutput;
-		double blurRadius = 2.7;
+        double blurRadius = 2.7;//2.7;
         double kernelSize = 2 * (blurRadius + 0.5) + 1;
         Mat blurOutput = new Mat();
         Imgproc.blur(blurInput, blurOutput, new Size(kernelSize, kernelSize));
-        
+
         // Convert to HSV and Threshold
         Mat hsvThresholdInput = blurOutput;
-		double[] hsvThresholdHue = {20.0, 180.0};
-		double[] hsvThresholdSaturation = {130.0, 255.0};
+        double[] hsvThresholdHue = {20.0, 90.0};//{20.0, 180.0};
+        double[] hsvThresholdSaturation = {130.0, 255.0};
         double[] hsvThresholdValue = {0.0, 255.0};
         Mat hsvOutput = new Mat();
         Imgproc.cvtColor(hsvThresholdInput, hsvOutput, Imgproc.COLOR_BGR2HSV);
-		Core.inRange(hsvOutput, new Scalar(hsvThresholdHue[0], hsvThresholdSaturation[0], hsvThresholdValue[0]),
-            new Scalar(hsvThresholdHue[1], hsvThresholdSaturation[1], hsvThresholdValue[1]), hsvOutput);
+        Core.inRange(hsvOutput, new Scalar(hsvThresholdHue[0], hsvThresholdSaturation[0], hsvThresholdValue[0]),
+                new Scalar(hsvThresholdHue[1], hsvThresholdSaturation[1], hsvThresholdValue[1]), hsvOutput);
 
-        //Find and draw contours
+        // Find and draw contours
         Imgcodecs.imwrite("HSV Output.jpg", hsvOutput);
```

Comment:
This code is run on the robot. You don't want to be writing a file every time the pipeline is run.

Reply:
Good catch, will remove